pact/provider/core/src/test/java/org/arquillian/algeron/pact/provider/core/ArgumentPatternMatcherTest.java

```diff
@@ -0,0 +1,36 @@
+package org.arquillian.algeron.pact.provider.core;
+
+import org.junit.Test;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ArgumentPatternMatcherTest {
+
+    @Test
+    public void should_extract_arguments_if_number() {
+        List<String> arguments = ArgumentPatternMatcher.arguments(Pattern.compile("I have (\\d+) cukes in my belly"), "I have 36 cukes in my belly");
+        assertThat(arguments).containsExactly("36");
+    }
+
+    @Test
+    public void should_extract_arguments_if_list() {
+        List<String> arguments = ArgumentPatternMatcher.arguments(Pattern.compile("The following animals: (.*)"), "The following animals: dog, cat, mouse");
+        assertThat(arguments).containsExactly("dog, cat, mouse");
```

Comment:
only once. Suppose you have this state: `I need a user called Alex with roles A, B, C`.
Then in provider side you'll have something like: `I need a user called (\\w+) with roles (.*)`.

Then the ArgumentMatcher will return a list containing two elements ["Alex", "A, B, C"]. Then these elements are casted to respective types depending on the state method, for example:

`registerUser(String name, Collection<String> roles)`